packages/byods/src/data-source-service/index.ts

```diff
@@ -0,0 +1,133 @@
+/* eslint-disable no-useless-catch */
+/* eslint-disable no-console */
+/* eslint-disable class-methods-use-this */
+/* eslint-disable import/no-extraneous-dependencies */
+
+import {v4 as uuidv4} from 'uuid';
+import {decodeJwt, JWTPayload} from 'jose';
+import DataSourceClient from '../data-source-client/index';
+
+export default class DataSourceService {
+  private dataSourceClient: DataSourceClient;
+  private timer: any;
+
+  /**
+   * Creates an instance of DataSourceService.
+   * @param {DataSourceClient} dataSourceClient - The  dataSourceClient instance to use for DataSourceService.
+   * @example
+   * const dataSourceService = new DataSourceService(dataSourceClient);
+   */
+
+  constructor(dataSourceClient: DataSourceClient) {
+    this.dataSourceClient = dataSourceClient;
+  }
+
+  /**
+   * This method refreshes the DataSource token using dataSourceId and tokenLifetimeMinute
```

Comment:
Add code examples in the ts-doc for this and the other public methods. This will generate and example section automatically in the reference doc

Reply:
**_Addressed!_**